app/frontend/src/lib/ui/components/searchCheckbox.js

```diff
@@ -0,0 +1,30 @@
+
+
+export const init = (container) => {
+  const checkboxes = container.getElementsByClassName('govuk-checkboxes__input');
+  const input = container.getElementsByClassName('search-input__input')[0];
+
+  input.addEventListener('input', (e) => {
+    filterCheckboxes(checkboxes, e.target);
+  });
+
+  input.addEventListener('change', (e) => {
+    filterCheckboxes(checkboxes, e.target);
+  });
+}
+
+export const filterCheckboxes = (checkboxes, input) => Array.from(checkboxes).map(checkbox => checkboxDisplay(checkbox, input));
+
+export const stringExistsInString = (original, input) => original.toUpperCase().indexOf(input.toUpperCase()) > -1;
+
+export const checkboxDisplay = (checkbox, input) => {
```

Comment:
Is the expectation from UX that this will behave how the subjects filter currently does? (i.e when selected the element goes to the top of the input). Not saying this needs to be done right now, but worth thinking about!

Reply:
good point, i hadnt accounted for this

Reply:
thinking about this, yes lets leave this for now. i think this behaviour is bad user experience and id like to discuss with chris first